Source/JavaScriptCore/b3/air/AirAllocateRegistersAndStackAndGenerateCode.cpp

```diff
@@ -599,7 +608,7 @@ void GenerateAndAllocateRegisters::generate(CCallHelpers& jit)
 
             bool needsToGenerate = ([&] () -> bool {
                 // FIXME: We should consider trying to figure out if we can also elide Mov32s
-                if (!(inst.kind.opcode == Move || inst.kind.opcode == MoveDouble))
+                if (!(inst.kind.opcode == Move || inst.kind.opcode == MoveDouble || (is32Bit() && inst.kind.opcode == Move32)))
```

Comment:
Is this right? I think we should not be conditioned on is32bit here, either it should work for 32bit or it shouldn't. Is this only an optimization?

Reply:
Maybe we can make this more obvious by splitting this out to a bool?